### ISSUE TITLE
Docker compose / Nginx: Cannot add manuals to equipment section

### ISSUE THREAD

**Issue by kingo55** [grocy/grocy-docker#15]
Whenever I add a manual PDF, Nginx throws 404 on the PUT request, like so:

![Screenshot_20190425_174702](https://user-images.githubusercontent.com/2361388/56718764-c2049380-6782-11e9-97d1-38cfa78b676f.png)

![Screenshot_20190425_174715](https://user-images.githubusercontent.com/2361388/56718774-c5981a80-6782-11e9-918a-76eb903a65e6.png)

![Screenshot_20190425_174918](https://user-images.githubusercontent.com/2361388/56718780-ca5cce80-6782-11e9-981c-b105657fa198.png)

Could be an issue with Docker-compose / Nginx config. I'm not really familiar with CRUD, but any ideas on how I can fix?

**Comment by berrnd** [comment]
I moved this issue to grocy/grocy-docker.

Seems to be the same as #11.

Still not using Docker myself, but I also run grocy exclusively on nginx/PHP-FPM and another nginx in front as a reverse proxy, so here is the full config of https://demo-en.grocy.info - maybe that helps...

Application host:
```
server {
	server_name demo-en.grocy.info;
	
	access_log /var/log/nginx/demo-en.grocy.info.access.log;
	error_log /var/log/nginx/demo-en.grocy.info.error.log;

	root /var/www/demo-en.grocy.info/public;

	location / {
		try_files $uri /index.php;
		
		location ~* \.(php|inc)$ {
			include fastcgi.conf;
			
			fastcgi_pass unix:/run/php/php7.2-fpm.sock;
			fastcgi_index index.php;
			fastcgi_intercept_errors on;
			fastcgi_buffer_size 16k;
			fastcgi_buffers 4 16k;
		}
	}

	location = /favicon.ico {
		log_not_found off;
		access_log off;
	}

	location = /robots.txt {
		log_not_found off;
		access_log off;
	}

	location ~ /\.ht {
		deny all;
	}
}
```

Reverse proxy:
```
server {
	listen 443 ssl http2;
	server_name demo-en.grocy.info;
	
	ssl_certificate /opt/centralcfg/sslcerts/letsencrypt/grocy.info/fullchain.cer;
	ssl_certificate_key /opt/centralcfg/sslcerts/letsencrypt/grocy.info/grocy.info.key;
	ssl_trusted_certificate /opt/centralcfg/sslcerts/letsencrypt/grocy.info/ca.cer;

	ssl_certificate /opt/centralcfg/sslcerts/letsencrypt/grocy.info_ecc/fullchain.cer;
	ssl_certificate_key /opt/centralcfg/sslcerts/letsencrypt/grocy.info_ecc/grocy.info.key;
	
	access_log /var/log/nginx/demo-en.grocy.info.access.log;
	error_log /var/log/nginx/demo-en.grocy.info.error.log;

	location / {
		proxy_set_header Host $host;
		proxy_set_header X-Real-IP $remote_addr;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header X-Forwarded-Proto $scheme;
		proxy_buffering off;
		proxy_request_buffering off;
		proxy_pass http://192.168.0.106/;
	}
}

server {
	server_name demo-en.grocy.info;
	
	access_log /var/log/nginx/demo-en.grocy.info.log;
	error_log /var/log/nginx/demo-en.grocy.info.log;
	
	return 301 https://$host$request_uri;
}
```

**Comment by talmai** [comment]
@berrnd was this resolved in some later version of grocy? would a more recent image fix this?

**Comment by berrnd** [comment]
No - the screenshot above shows a 404 page of nginx, so the request have not hit grocy - I assume something about URL rewriting could be the problem...

**Comment by berrnd** [comment]
I'm not a nginx config expert, but you defined the URL rewriting here:
https://github.com/grocy/grocy-docker/blob/6e57e4652ca5681376e7de87c98e71438c7f7ffe/docker_nginx/common.conf#L6

The following definitely works on all of my installations (see full config above), maybe try changing that... (?)
```
try_files $uri /index.php;
```

**Comment by jgoett154** [comment]
I was having this same issue with the 404 error. After a lot of poking around and binding a volume for the nginx log I determined it seems to be caused by a lack of permissions for /var/tmp/nginx/client_body. I am unsure why this is occurring exactly but what appears to happen is a temp file is created in that directory for file uploads and for whatever reason fails to open. Below is the log line that occurs when an upload happens. The reason for the 404 though is caused by a lack of a 50x.html file which would normally be displayed when the error below occurs. I changed the worker processes to be run as root which seems to be working (though obviously not ideal). I am not great at docker or nginx so I am not entirely sure how to go about resolving this. But hopefully this info assists with it being taken care of. I will continue trying different things to get it working but I don't have a huge amount of time during the week.

`2019/05/12 23:00:06 [crit] 6#6: *1 open() "/var/tmp/nginx/client_body/0000000001" failed (13: Permission denied), client: 172.18.0.1, server: _, request: "PUT /api/files/productpictures/Y29sYS5qcGVn HTTP/1.1", host: "localhost:8083", referrer: "http://localhost:8083/api"`
